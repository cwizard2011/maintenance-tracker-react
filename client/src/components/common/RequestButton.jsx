import React from 'react';
import PropTypes from 'prop-types';

const RequestButton = (props) => {
  const {
    handleClick, iconName, actionName, buttonClass
  } = props;
  return (
    <span className="right-text">
      <button
        type="button"
        className={buttonClass}
        onClick={handleClick}
      >
        <ion-icon name={iconName} />
        {actionName}
      </button>
    </span>
  );
};
RequestButton.propTypes = {
  handleClick: PropTypes.func,
  iconName: PropTypes.string,
  actionName: PropTypes.string,
  buttonClass: PropTypes.string
};
export default RequestButton;
