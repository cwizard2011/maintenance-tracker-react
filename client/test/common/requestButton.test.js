import React from 'react';
import { shallow } from 'enzyme';
import RequestButton from '../../src/components/common/RequestButton';

describe('Request button', () => {
  it('render request button correctly', () => {
    const wrapper = shallow(<RequestButton />);
    expect(wrapper).toMatchSnapshot();
  });
});
