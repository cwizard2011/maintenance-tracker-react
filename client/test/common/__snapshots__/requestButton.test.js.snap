// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request button render request button correctly 1`] = `
<span
  className="right-text"
>
  <button
    type="button"
  >
    <ion-icon />
  </button>
</span>
`;
